src/array/array.F90

```diff
@@ -76,7 +78,17 @@ function clone( from )
     class(array_t), pointer :: clone
     class(array_t), intent(in) :: from
 
-    allocate( clone, SOURCE = from )
+    select type(from)
+    type is(array_t)
+      ! for concrete base class, copy data_ from source
+      allocate( clone )
+      if( associated( from%data_ ) ) then
+        allocate( clone%data_, SOURCE = from%data_ )
+      end if
+    class default
+      ! provide default clone functionality for subclasses
+      allocate( clone, SOURCE = from )
```

Comment:
Is this still to be done?

Reply:
no, this is the default functionality - it creates a copy of `from`. The only reason a subclass would need to override this function is if it wanted 'non-standard' copy behavior (like copying the target of a pointer data member instead of copying just the pointer reference)